src/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSystemSecurity.cs

```diff
@@ -0,0 +1,291 @@
+                    break;
+
+                default:
+                    break;
+            }
+
+            return exception;
+        }
+
+        public sealed override AccessRule AccessRuleFactory(
+            IdentityReference identityReference,
+            int accessMask,
+            bool isInherited,
+            InheritanceFlags inheritanceFlags,
+            PropagationFlags propagationFlags,
+            AccessControlType type)
+        {
+            return new FileSystemAccessRule(
+                identityReference,
+                accessMask,
+                isInherited,
+                inheritanceFlags,
+                propagationFlags,
+                type);
+        }
+
+        public sealed override AuditRule AuditRuleFactory(
+            IdentityReference identityReference,
+            int accessMask,
+            bool isInherited,
+            InheritanceFlags inheritanceFlags,
+            PropagationFlags propagationFlags,
+            AuditFlags flags)
+        {
+            return new FileSystemAuditRule(
+                identityReference,
+                accessMask,
+                isInherited,
+                inheritanceFlags,
+                propagationFlags,
+                flags);
+        }
+
+        internal AccessControlSections GetAccessControlSectionsFromChanges()
+        {
+            AccessControlSections persistRules = AccessControlSections.None;
+            if (AccessRulesModified)
+                persistRules = AccessControlSections.Access;
+            if (AuditRulesModified)
+                persistRules |= AccessControlSections.Audit;
+            if (OwnerModified)
+                persistRules |= AccessControlSections.Owner;
+            if (GroupModified)
+                persistRules |= AccessControlSections.Group;
+            return persistRules;
+        }
+
+        internal void Persist(string fullPath)
+        {
+            WriteLock();
+
+            try
+            {
+                AccessControlSections persistRules = GetAccessControlSectionsFromChanges();
+                base.Persist(fullPath, persistRules);
+                OwnerModified = GroupModified = AuditRulesModified = AccessRulesModified = false;
+            }
+            finally
+            {
+                WriteUnlock();
+            }
+        }
+
+        internal void Persist(SafeFileHandle handle, string fullPath)
+        {
+            WriteLock();
+
+            try
+            {
+                AccessControlSections persistRules = GetAccessControlSectionsFromChanges();
+                Persist(handle, persistRules);
+                OwnerModified = GroupModified = AuditRulesModified = AccessRulesModified = false;
+            }
+            finally
+            {
+                WriteUnlock();
+            }
+        }
+
+        public void AddAccessRule(FileSystemAccessRule rule)
+        {
+            base.AddAccessRule(rule);
+            // PersistIfPossible();
```

Comment:
Should this commented line be removed?

Reply:
Probably, but I would want to look at the history a bit more and do it in another PR.